src/webviews/Document/DocumentToolbar.tsx

```diff
@@ -3,115 +3,60 @@
-import { useDocumentDispatcher, useDocumentState } from './state/DocumentContext';
-
-type AlertDialogProps = {
-    open: boolean;
-    setOpen: (open: boolean) => void;
-    doAction: () => Promise<void>;
-};
-
-const AlertDialog = ({ open, setOpen, doAction }: AlertDialogProps) => {
-    return (
-        <Dialog modalType="alert" open={open} onOpenChange={(_event, data) => setOpen(data.open)}>
-            <DialogSurface>
-                <DialogBody>
-                    <DialogTitle>Attention</DialogTitle>
-                    <DialogContent>
-                        <div>Your document has unsaved changes. If you continue, these changes will be lost.</div>
-                        <div>Are you sure you want to continue?</div>
-                    </DialogContent>
-
-                    <DialogActions>
-                        <Button appearance="secondary" onClick={() => void doAction()}>
-                            Continue
-                        </Button>
-
-                        <DialogTrigger disableButtonEnhancement>
-                            <Button appearance="primary">Close</Button>
-                        </DialogTrigger>
-                    </DialogActions>
-                </DialogBody>
-            </DialogSurface>
-        </Dialog>
-    );
-};
+import { useDocumentState } from './state/DocumentContext';
 
 const ToolbarDividerTransparent = () => {
     return <div style={{ padding: '4px' }} />;
 };
 
-export const DocumentToolbar = () => {
-    const state = useDocumentState();
-    const dispatcher = useDocumentDispatcher();
+export type DocumentToolbarProps = {
+    onSave: () => Promise<void>;
+    onEdit: () => Promise<void>;
+    onRefresh: () => Promise<void>;
+};
 
-    const [open, setOpen] = useState(false);
-    const [doAction, setDoAction] = useState<() => Promise<void>>(() => async () => {});
+export const DocumentToolbar = (props: DocumentToolbarProps) => {
+    const state = useDocumentState();
 
     const inProgress = state.isSaving || state.isRefreshing;
     const hasDocumentInDB = state.documentId !== undefined;
-    const isDirty = state.isDirty;
     const isReadOnly = state.mode === 'view';
+    const isMac = navigator.platform.toLowerCase().includes('mac');
 
-    const onSaveRequest = () => {
-        // Save document to the database
-        void dispatcher.saveDocument(state.currentDocumentContent);
-    };
-
-    const onEditRequest = () => {
-        // Open document for editing
-        void dispatcher.setMode('edit');
-    };
-
-    const onRefreshRequest = () => {
-        // Reload original document from the database
-        if (state.isDirty) {
-            setOpen(true);
-            setDoAction(() => async () => {
-                setOpen(false);
-                await dispatcher.refreshDocument();
-            });
-        } else {
-            void dispatcher.refreshDocument();
-        }
-    };
+    const onSaveHotkeyTitle = isMac ? 'Cmd+S' : 'Ctrl+S';
```

Comment:
Nit: on Mac the special keys are usually represented by their symbols:
* Command: ⌘ (U+2318)
* Shift: ⇧ (U+21E7)
<img width="261" alt="image" src="https://github.com/user-attachments/assets/b3fbbba9-37af-4892-b32e-4188a3010266">

Reply:
I've asked you recently how it works on Mac, you didn't response  :)